x-pack/plugin/ml/src/internalClusterTest/java/org/elasticsearch/xpack/ml/integration/TrainedModelCRUDIT.java

```diff
@@ -0,0 +1,108 @@
+            "Z5nVQIAAAACAAAAUEsBAgAAAAAICAAAAAAAAFzqQQQ0AAAANAAAABQAAAAAAAAAAAAAAAAAAAAAAHNpbXBsZW1vZGVsL2RhdGEucGts" +
+            "UEsBAgAAFAAICAgAAAAAAE0EGH3yAAAAlwEAAB0AAAAAAAAAAAAAAAAAhAAAAHNpbXBsZW1vZGVsL2NvZGUvX190b3JjaF9fLnB5UEs" +
+            "BAgAAFAAICAgAAAAAAJ7FExAzAQAABQMAACcAAAAAAAAAAAAAAAAAAgIAAHNpbXBsZW1vZGVsL2NvZGUvX190b3JjaF9fLnB5LmRlYn" +
+            "VnX3BrbFBLAQIAAAAACAgAAAAAAABtLwlXBAAAAAQAAAAZAAAAAAAAAAAAAAAAAMMDAABzaW1wbGVtb2RlbC9jb25zdGFudHMucGtsU" +
+            "EsBAgAAAAAICAAAAAAAANGeZ1UCAAAAAgAAABMAAAAAAAAAAAAAAAAAFAQAAHNpbXBsZW1vZGVsL3ZlcnNpb25QSwYGLAAAAAAAAAAe" +
+            "Ay0AAAAAAAAAAAAFAAAAAAAAAAUAAAAAAAAAagEAAAAAAACSBAAAAAAAAFBLBgcAAAAA/AUAAAAAAAABAAAAUEsFBgAAAAAFAAUAagE" +
+            "AAJIEAAAAAA==";
+    static final long RAW_MODEL_SIZE; // size of the model before base64 encoding
+    static {
+        RAW_MODEL_SIZE = Base64.getDecoder().decode(BASE_64_ENCODED_MODEL).length;
+    }
+
+    @Before
+    public void createComponents() throws Exception {
+        waitForMlTemplates();
+    }
+
+    public void testPutTrainedModelAndDefinition() {
+        String modelId = "pytorch-model-put";
+        TrainedModelConfig config = client().execute(
+            PutTrainedModelAction.INSTANCE,
+            new PutTrainedModelAction.Request(
+                TrainedModelConfig.builder()
+                    .setModelType(TrainedModelType.PYTORCH)
+                    .setInferenceConfig(
+                        new BertPassThroughConfig(
+                            new VocabularyConfig(
+                                InferenceIndexConstants.nativeDefinitionStore(),
+                                modelId + "_vocab"
+                            ),
+                            new BertTokenization(null, false, null)
+                        )
+                    )
+                    .setModelId(modelId)
+                    .build(),
+                false
+            )
+        ).actionGet().getResponse();
+
+        assertThat(config.getLocation(), isA(IndexLocation.class));
+        assertThat(((IndexLocation) config.getLocation()).getIndexName(), equalTo(InferenceIndexConstants.nativeDefinitionStore()));
+        client().execute(
+            PutTrainedModelDefinitionPartAction.INSTANCE,
+            new PutTrainedModelDefinitionPartAction.Request(modelId, new BytesArray(BASE_64_ENCODED_MODEL), 0, RAW_MODEL_SIZE, 1)
+        ).actionGet();
+
+        assertThat(
+            client().admin()
+                .indices()
+                .prepareGetIndex()
+                .addIndices(InferenceIndexConstants.nativeDefinitionStore())
+                .get()
+                .indices().length,
+            equalTo(1)
+        );
+
+        client().execute(
+            DeleteTrainedModelAction.INSTANCE,
+            new DeleteTrainedModelAction.Request(modelId)
+        ).actionGet();
+    }
```

Comment:
nit: Can you add a test using the count API to assert there are `0` documents in `InferenceIndexConstants.nativeDefinitionStore()` after the model is deleted